keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -320,28 +344,54 @@ class ApplicationService(
-      ?: throw InvalidSystemStateException("Loading artifact version $version failed for known artifact $artifact.")
 
+    val artifactSupplier = artifactSuppliers.supporting(artifact.type)
+    val artifactInstance = getArtifactInstance(artifact, version)
+            ?: throw InvalidSystemStateException("Loading artifact version $version failed for known artifact $artifact.")
     return ArtifactVersionSummary(
-      version = version,
-      environments = environments,
-      displayName = artifactSupplier.getVersionDisplayName(artifactInstance),
-      createdAt = artifactInstance.createdAt,
-
-      // first attempt to use the artifact metadata fetched from the DB, then fallback to the default if not found
-      build = artifactInstance.buildMetadata
-        ?: artifactSupplier.parseDefaultBuildMetadata(artifactInstance, artifact.versioningStrategy),
-      git = artifactInstance.gitMetadata
-        ?: artifactSupplier.parseDefaultGitMetadata(artifactInstance, artifact.versioningStrategy)
-    )
-  }
+        version = version,
+        environments = environments,
+        displayName = artifactSupplier.getVersionDisplayName(artifactInstance),
+        createdAt = artifactInstance.createdAt,
+
+        // first attempt to use the artifact metadata fetched from the DB, then fallback to the default if not found
+        build = artifactInstance.buildMetadata
+          ?: artifactSupplier.parseDefaultBuildMetadata(artifactInstance, artifact.versioningStrategy),
+        git = artifactInstance.gitMetadata
+          ?: artifactSupplier.parseDefaultGitMetadata(artifactInstance, artifact.versioningStrategy)
+      )
+    }
 
   fun getApplicationEventHistory(application: String, limit: Int) =
     repository.applicationEventHistory(application, limit)
 
   private fun ConstraintState.toConstraintSummary() =
     StatefulConstraintSummary(type, status, createdAt, judgedBy, judgedAt, comment, attributes)
+
+  private fun getArtifactInstance(
+    artifact: DeliveryArtifact,
+    version: String
+  ): PublishedArtifact? {
+    val releaseStatus = repository.getReleaseStatus(artifact, version)
+    return repository.getArtifactInstance(artifact.name, artifact.type, version, releaseStatus)
+  }
+
+  // Generating a stash diff link between source and target versions (the order does matter!)
+  private fun generateDiffLink (newerGitMetadata: GitMetadata?, olderGitMetadata: GitMetadata?) : String? {
+    return if (newerGitMetadata !=null && olderGitMetadata != null) {
+      "https://stash.corp.netflix.com/projects/${newerGitMetadata.project}/repos/${newerGitMetadata.repo?.name}/compare/diff?" +
+        "targetBranch=${olderGitMetadata.commitInfo?.sha}&sourceBranch=${newerGitMetadata.commitInfo?.sha}"
+    } else {
+      null
+    }
+  }
+
+  private fun getPotentialSummary (deliveryConfig: DeliveryConfig, environmentName: String, artifactReference: String, version: String) =
```

Comment:
This is just wrapping a single call with the same number of parameters. What's the goal of this function?

Reply:
reverted it and deleted this function